malloc.c gives compiler warnings under multiplicity

I use gcc 11.3.0. with function -DPERL_CORE -DPERL_IS_MINIPERL -DUSE_ITHREADS
https://github.com/Perl/perl5/commit/932a38d1a532eec8faccdcb6e4ce4d023fbf70be
Are these messages correct?
```
locale.c: In function 'S_my_langinfo_i':
locale.c:4144:14: warning: '___lc_codepage_func' redeclared without dllimport attribute: previous dllimport ignored [-Wattributes]
 4144 |         UINT ___lc_codepage_func(void);
      |              ^~~~~~~~~~~~~~~~~~~
malloc.c:242: warning: "croak" redefined
  242 | #    define croak       Perl_croak_nocontext
      |
In file included from perl.h:6217,
                 from malloc.c:240:
embed.h:114: note: this is the location of the previous definition
  114 | #define croak(...)              Perl_croak(aTHX_ __VA_ARGS__)
      |
malloc.c:244: warning: "warn" redefined
  244 | #    define warn        Perl_warn_nocontext
      |
In file included from perl.h:6217,
                 from malloc.c:240:
embed.h:743: note: this is the location of the previous definition
  743 | #define warn(...)               Perl_warn(aTHX_ __VA_ARGS__)
      |
```
https://github.com/Perl/perl5/pull/20289

--- a/malloc.c
+++ b/malloc.c
@@ -239,13 +239,9 @@
 #define PERL_IN_MALLOC_C
 #include "perl.h"
 #if defined(MULTIPLICITY)
-#    define croak	Perl_croak_nocontext
 #    define croak2	Perl_croak_nocontext
-#    define warn	Perl_warn_nocontext
-#    define warn2	Perl_warn_nocontext
 #else
 #    define croak2	croak
-#    define warn2	warn
 #endif
 #ifdef USE_ITHREADS
 #     define PERL_MAYBE_ALIVE	PL_thr_key
@@ -1708,7 +1704,7 @@ morecore(int bucket)
 #endif
         if (bucket == sizeof(MEM_SIZE)*8*BUCKETS_PER_POW2) {
             MALLOC_UNLOCK;
-            croak("%s", "Out of memory during ridiculously large request");
+            croak2("%s", "Out of memory during ridiculously large request");
         }
         if (bucket > max_bucket)
             max_bucket = bucket;

